performance_manager/lib/rt_vehicle_positions.py

```diff
@@ -0,0 +1,277 @@
+import logging
+
+import pandas
+import numpy
+import sqlalchemy
+import re
+import datetime
+
+from sqlalchemy import select
+from sqlalchemy import update
+from sqlalchemy import insert
+from sqlalchemy import delete
+from sqlalchemy.orm import sessionmaker
+
+from typing import Optional
+from typing import Union
+from typing import List
+
+from .s3_utils import read_parquet
+from .postgres_utils import VehiclePositionEvents
+
+
+def get_vp_dataframe(to_load: Union[str, List[str]]) -> pandas.DataFrame:
+    """
+    return a dataframe from a vehicle position parquet file (or list of files)
+    with expected columns
+    """
+    vehicle_position_cols = [
+        "current_status",
+        "current_stop_sequence",
+        "stop_id",
+        "vehicle_timestamp",
+        "direction_id",
+        "route_id",
+        "start_date",
+        "start_time",
+        "vehicle_id",
+    ]
+
+    return read_parquet(to_load).loc[:, vehicle_position_cols]
+
+
+def start_time_to_seconds(
+    time: Optional[str],
+) -> Optional[float]:
+    """
+    transform time string in HH:MM:SS format to seconds integer
+    """
+    if time is None:
+        return time
+    (hour, min, sec) = time.split(":")
+    return int(hour) * 3600 + int(min) * 60 + int(sec)
+
+
+def transform_vp_dtyes(df: pandas.DataFrame) -> pandas.DataFrame:
+    """
+    ingest dataframe of vehicle position data from parquet file and transform
+    column datatypes
+    """
```

Comment:
Initial dataframe transformation function. Main purpose is to convert datatypes, but also add hash and drops all records with duplicate hash and timestamp values.